app/assets/javascripts/static_pages.js

```diff
@@ -21,8 +24,128 @@ $(document).ready(function(){
     
     var service = new google.maps.places.PlacesService(map);
       service.nearbySearch(request, callback);
+
+    bind_controls(map);
   }
 
+// Goal: I must bind and setup search control for the map.param: map - the Google map object
+
+var bind_controls = function(mapParam) {
+  // get the container for the search control and bind and event to it on submit
+  var controlContainer = $('#control_container')[0];
+  google.maps.event.addDomListener(controlContainer, 'submit', function(e) {
+    e.preventDefault();
+    search(mapParam);
+  });
+
+  // get the search button and bind a click event to it for searching
+  var searchButton = $('#map_search_submit')[0];
+  google.maps.event.addDomListener(searchButton, 'click', function(e) {
+    e.preventDefault();
+    search(map);
```

Comment:
it should be `search(mapParam)` here, as well as on line 49: `mapParam.controls...`